Dashboard/app/js/lib/controllers/survey-controllers.js

```diff
@@ -224,180 +218,170 @@ FLOW.projectControl = Ember.ArrayController.create({
-    }));
+    this.set('content', FLOW.store.filter(FLOW.SurveyGroup, () => true));
   },
 
-  setCurrentProject: function(project) {
+  setCurrentProject(project) {
     this.set('currentProject', project);
-    window.scrollTo(0,0);
+    window.scrollTo(0, 0);
   },
 
   /* return true if the given SurveyGroup's has the data cleaning permission
    * associated with it, or if one of the ancestors or descendants of the SurveyGroup
    * has data cleaning permission associated with it.  In the case of descendants we
    * return true in order to be able to browse to the descendant */
-  dataCleaningEnabled: function(surveyGroup) {
-    var permissions = FLOW.currentUser.get('pathPermissions');
-    var keyedSurvey;
+  dataCleaningEnabled(surveyGroup) {
+    const permissions = FLOW.currentUser.get('pathPermissions');
 
-    for (var key in permissions) {
-      if(permissions[key].indexOf("DATA_CLEANING") > -1){
+    Object.keys(permissions).forEach((key) => {
+      if (permissions[key].indexOf('DATA_CLEANING') > -1) {
         // check key against survey group
-        if(surveyGroup.get('keyId') === +key) {
+        if (surveyGroup.get('keyId') === +key) {
           return true;
         }
 
         // check ancestors to for matching permission from higher level in hierarchy
-        var ancestorIds = surveyGroup.get('ancestorIds');
+        const ancestorIds = surveyGroup.get('ancestorIds');
         if (ancestorIds === null) {
           return false;
-        } else {
-          for(var i = 0; i < ancestorIds.length; i++){
-            if(ancestorIds[i] === +key) {
-              return true;
-            }
+        }
+        for (let i = 0; i < ancestorIds.length; i++) {
+          if (ancestorIds[i] === +key) {
+            return true;
           }
         }
 
+
         // finally check for all descendents that may have surveyGroup.keyId in their
         // ancestor list otherwise will not be able to browse to them.
-        keyedSurvey = FLOW.store.find(FLOW.SurveyGroup, key);
+        const keyedSurvey = FLOW.store.find(FLOW.SurveyGroup, key);
         if (keyedSurvey) {
-          var keyedAncestorIds = keyedSurvey.get('ancestorIds');
+          const keyedAncestorIds = keyedSurvey.get('ancestorIds');
           if (keyedAncestorIds === null) {
             return false;
-          } else {
-            for (var j = 0; j < keyedAncestorIds.length; j++) {
-              if(keyedAncestorIds[j] === surveyGroup.get('keyId')) {
-                return true;
-              }
+          }
+          for (let j = 0; j < keyedAncestorIds.length; j++) {
+            if (keyedAncestorIds[j] === surveyGroup.get('keyId')) {
+              return true;
             }
           }
         }
       }
-    }
+    });
+
     return false;
   },
 
   /* Computed properties */
-  breadCrumbs: Ember.computed(function() {
-    var result = [];
-    var currentProject = this.get('currentProject');
+  breadCrumbs: Ember.computed(function () {
```

Comment:
@finnfiddle like here..

Reply:
@muloem so the answer is that it depends whether `this` has been used inside the function. Inside normal es5 functions `this` refers to the function itself, but in arrow functions `this` refers to the context that the function was defined in. Eg:
```
var self = this;
function oldSchool() {
  assert(this === self, false);
}
const newSchool = () => {
  assert(this === self, true);
}
```
Its called lexical scoping if you want to read up on it: https://hackernoon.com/javascript-es6-arrow-functions-and-lexical-this-f2a3e2a5e8c4

So Eslint has automatically converted the ones it can to arrow functions and the other contain `this` it has left as is.

Reply:
can I resolve @muloem?